jdaviz/configs/mosviz/plugins/parsers.py

```diff
@@ -87,6 +89,35 @@ def _fields_from_ecsv(fname, fields, delimiter=","):
     return parsed_fields
 
 
+@data_parser_registry("mosviz-link-data")
+def link_data_in_table(app, data_obj=None, data_labels=None):
+    """
+    Batch links data in the mosviz table viewer
+
+    Parameters
+    ----------
+    app : `~jdaviz.app.Application`
+        The application-level object used to reference the viewers.
+    data_obj : None
+        Passed in in order to use the data_parser_registry, otherwise
+        not used.
+    data_labels : None
+        Not used.
+    """
+    mos_data = app.session.data_collection['MOS Table']
+
+    # Loop through mos table rows and link Spectra 1D and Spectra 2D
+    # data objects. This speeds up performance significantly over linking
+    # data when loaded into the application.
+    for index in range(0, len(mos_data.get_component('1D Spectra').data)):
+        spec_1d = mos_data.get_component('1D Spectra').data[index]
+        spec_2d = mos_data.get_component('2D Spectra').data[index]
+
+        wc_spec_1d = app.session.data_collection[spec_1d].world_component_ids
+        wc_spec_2d = app.session.data_collection[spec_2d].world_component_ids
+        app.session.data_collection.add_link(LinkSame(wc_spec_1d[0], wc_spec_2d[0]))
```

Comment:
This is a better linking approach than what is in main at the moment, but you can then get significantly better performance for this step if you also use the context manager I used in https://github.com/spacetelescope/jdaviz/pull/782 - could you test this out?

Reply:
Now that #782 is merged you can encapsulate the whole for loop in the delay context manager

Reply:
OK, I see it now. The gain in speed for using the delay context manager here (close to an order of magnitude for `link_table_data()` on its own with 20 spectra + images) is similar to what we found with a different approach of appending the `LinkSame` object from each iteration of this loop to a list and waiting until after the loop to call `add_link()` on that list.

Combining these approaches is even faster. Are any of these strategies better or worse from a `glue` perspective?

Reply:
Combining the two approaches makes sense from a glue perspective!